File: src/main/java/toby/helpers/URLHelper.java
package toby.helpers;

import java.net.URI;
import java.net.URL;

public class URLHelper {

    public static boolean isValidURL(String url) {
        /* Try creating a valid URL */
        try {
            new URL(url).toURI();
            return true;
        }
        // If there was an Exception
        // while creating URL object
        catch (Exception e) {
            return false;
        }
    }

    public static URI fromUrlString(String url){
        /* Try creating a valid URL */
        try {
            return new URL(url).toURI();
        }
        // If there was an Exception
        // while creating URL object
        catch (Exception e) {
            return null;
        }
    }
}

File: src/main/java/toby/helpers/URLHelper.kt
package toby.helpers

import java.net.URI
import java.net.URL
import java.util.*

object URLHelper {
    fun isValidURL(url: String?): Boolean {
        /* Try creating a valid URL */
        try {
            URL(url).toURI()
            return true
        } // If there was an Exception
        // while creating URL object
        catch (e: Exception) {
            return false
        }
    }

    fun fromUrlString(url: String?): Optional<URI> {
        /* Try creating a valid URL */
        return try {
            Optional.of(URL(url).toURI())
        } // If there was an Exception
        // while creating URL object
        catch (e: Exception) {
            Optional.empty()
        }
    }
}
